doc/adr/0003-directory-structure.md

```diff
@@ -0,0 +1,39 @@
+# 3. Define initial testing infrastructure
+
+Date: 2020-12-01
+
+## Status
+
+Accepted
+
+## Context
+
+The will be many services contributed by multiple people. We need a common way of structuring the services
+so it's easy for the user to pick up any one service.
+
+## Decision
+
+Each service will be its own library. All services will be in the services directory.
+Each service will contain at least an include folder and a mbed_lib.json.
+
+An example service called `example` would look like this:
+
+```
+services/ble-service-example/include/example.h
```

Comment:
That's not in line with what's in mbed-os at the moment which is `include/library-name` to include the header files with `library-name/header.h`.